middleware/csrf/csrf.go

```diff
@@ -243,10 +257,36 @@ func isFromCookie(extractor any) bool {
 	return reflect.ValueOf(extractor).Pointer() == reflect.ValueOf(FromCookie).Pointer()
 }
 
+// originMatchesHost checks that the origin header matches the host header
+// returns an error if the origin header is not present or is invalid
+// returns nil if the origin header is valid
+func originMatchesHost(c fiber.Ctx, trustedOrigins []string) error {
+	origin := c.Get(fiber.HeaderOrigin)
+	if origin == "" {
+		return errNoOrigin
+	}
+
+	originURL, err := url.Parse(origin)
+	if err != nil {
+		return ErrBadReferer
```

Comment:
Any way we can test this line?

Reply:
Invalid URL I suppose. Although it seems unlikely in practice because the browser would somehow need to navigate to an invalid URL then make a request and set that invalid URL in the header.

Reply:
Okay I used an invalid URL in the header. Covered.